src/kernel.cpp

```diff
@@ -100,6 +106,26 @@ void cvk_kernel::set_image_metadata(cl_uint index, const void* image) {
     }
 }
 
+void cvk_kernel::set_sampler_metadata(cl_uint index, const void* sampler) {
+    if (!m_sampler_metadata) {
+        return;
+    }
+    auto md = m_sampler_metadata->find(index);
+    if (md != m_sampler_metadata->end()) {
+        auto apisampler = *reinterpret_cast<const cl_sampler*>(sampler);
+        auto offset = md->second;
+        auto sampler = icd_downcast(apisampler);
+        uint32_t sampler_mask = (sampler->normalized_coords()
+                                     ? clspv::CLK_NORMALIZED_COORDS_TRUE
+                                     : clspv::CLK_NORMALIZED_COORDS_FALSE) |
+                                (sampler->filter_mode() == CL_FILTER_NEAREST
+                                     ? clspv::CLK_FILTER_NEAREST
+                                     : clspv::CLK_FILTER_LINEAR);
```

Comment:
Seems like this is missing the addressing mode.

Reply:
Yes, I was a bit lazy here. As it is not used by clspv, I did not put it here.

Reply:
Done